File: blur_screens.py
"""
Script de floutage des zones sensibles des screenshots pour la landing page.
"""
from PIL import Image, ImageFilter, ImageDraw
import shutil, os

SRC = r"C:\Users\tomro\OneDrive\Bureau\ScreensLanding"
DST = r"C:\Users\tomro\OneDrive\Bureau\APPsuivichantier\public"

FILES = {
    "image-1778852354493.png": "screen-planning.png",
    "image-1778852389340.png": "screen-profil.png",
    "image-1778852403008.png": "screen-rapport.png",
    "image-1778852412956.png": "screen-etapes.png",
    "image-1778852420710.png": "screen-dashboard.png",
}

def blur_region(img, box, radius=25):
    """Floute une région rectangulaire (left, top, right, bottom)."""
    region = img.crop(box)
    for _ in range(10):
        region = region.filter(ImageFilter.GaussianBlur(radius))
    img.paste(region, box[:2])
    return img

def solid_region(img, box, color=(220, 220, 220)):
    """Couvre une région par un rectangle solide."""
    draw = ImageDraw.Draw(img)
    draw.rectangle(box, fill=color)
    return img

def process(src_name, dst_name, regions):
    src_path = os.path.join(SRC, src_name)
    dst_path = os.path.join(DST, dst_name)
    img = Image.open(src_path).convert("RGB")
    w, h = img.size
    print(f"{dst_name} - {w}x{h}px")
    for box in regions:
        # box en proportions (left%, top%, right%, bottom%)
        left   = int(box[0] * w)
        top    = int(box[1] * h)
        right  = int(box[2] * w)
        bottom = int(box[3] * h)
        img = blur_region(img, (left, top, right, bottom))
    img.save(dst_path, quality=95)
    print(f"  OK : {dst_path}")

# ── Planning desktop ──────────────────────────────────────────────────────────
# Zones : "ROMAND tom / Admin" (haut droite) + ligne des noms des techs
process("image-1778852354493.png", "screen-planning.png", [
    (0.67, 0.00, 0.86, 0.08),   # "ROMAND tom / Admin"
    (0.19, 0.41, 0.97, 0.50),   # avatars (photos) + noms techniciens
])

# ── Profil mobile ─────────────────────────────────────────────────────────────
# Pas de nom/email visible — pas de floutage nécessaire
shutil.copy(os.path.join(SRC, "image-1778852389340.png"),
            os.path.join(DST, "screen-profil.png"))
print("screen-profil.png - copie tel quel")

# ── Rapport mobile ────────────────────────────────────────────────────────────
# Zones : header client ("Jean Dupont / 12 Rue des acacias") + auteur "Tom ROMAND"
src_path = os.path.join(SRC, "image-1778852403008.png")
dst_path = os.path.join(DST, "screen-rapport.png")
img = Image.open(src_path).convert("RGB")
w, h = img.size
print(f"screen-rapport.png - {w}x{h}px")
# header flou (client name + adresse + progress bar)
img = blur_region(img, (int(0.00*w), int(0.00*h), int(1.00*w), int(0.20*h)))
# "Tom ROMAND + date" en rectangle solide
img = solid_region(img, (int(0.00*w), int(0.24*h), int(1.00*w), int(0.31*h)), color=(235, 235, 235))
img.save(dst_path, quality=95)
print(f"  OK : {dst_path}")

# ── Étapes mobile ─────────────────────────────────────────────────────────────
# Zones : header client ("Jean Dupont / 12 Rue des acacias")
process("image-1778852412956.png", "screen-etapes.png", [
    (0.06, 0.03, 0.90, 0.16),   # "Jean Dupont" + "Dupont" grand + adresse
])

# ── Dashboard mobile ──────────────────────────────────────────────────────────
# Pas de donnée sensible - copie directe
shutil.copy(os.path.join(SRC, "image-1778852420710.png"),
            os.path.join(DST, "screen-dashboard.png"))
print("screen-dashboard.png - copie tel quel")

print("\nTous les screens traites et copies dans public/")

File: scripts/blur-screens.mjs
/**
 * blur-screens.mjs
 * Usage : node scripts/blur-screens.mjs <dossier_sources>
 * Ex    : node scripts/blur-screens.mjs C:\Users\tomro\Desktop\screens_bruts
 *
 * Lit les 5 screenshots depuis le dossier source, floute les zones sensibles,
 * et enregistre les résultats dans public/ (prêts pour la landing page).
 */

import sharp from 'sharp'
import path  from 'path'
import fs    from 'fs'

const src = process.argv[2] ?? 'C:\\Users\\tomro\\Desktop\\screens_bruts'
const dst = path.resolve('public')

if (!fs.existsSync(src)) {
  console.error(`❌ Dossier source introuvable : ${src}`)
  process.exit(1)
}

// ── Helper : floute une liste de rectangles (en % de la taille de l'image)
async function blurRegions(inputPath, outputPath, regions) {
  const img    = sharp(inputPath)
  const meta   = await img.metadata()
  const W      = meta.width
  const H      = meta.height

  // Convertit les % en pixels, floute chaque région, recompose
  let pipeline = sharp(inputPath)

  const overlays = await Promise.all(
    regions.map(async ({ x, y, w, h }) => {
      const px = Math.round(x * W)
      const py = Math.round(y * H)
      const pw = Math.round(w * W)
      const ph = Math.round(h * H)

      // Extrait la zone, la floute fortement, la renvoie comme overlay
      const blurred = await sharp(inputPath)
        .extract({ left: px, top: py, width: pw, height: ph })
        .blur(20)
        .toBuffer()

      return { input: blurred, left: px, top: py }
    })
  )

  await pipeline.composite(overlays).toFile(outputPath)
  console.log(`✅ ${path.basename(outputPath)} — ${regions.length} zone(s) floutée(s)`)
}

// ── Définition des zones à flouter par écran (coordonnées en % de l'image)

const SCREENS = [

  // ── Dashboard mobile — relativement clean, on floute juste les noms de chantiers
  {
    src: 'screen-dashboard.png',
    dst: 'screen-dashboard.png',
    regions: [
      // Description anomalie "Dijoncteur en surchauffe"
      { x: 0.35, y: 0.74, w: 0.55, h: 0.04 },
    ],
  },

  // ── Profil mobile — nom + email
  {
    src: 'screen-profil.png',
    dst: 'screen-profil.png',
    regions: [
      // Nom "Tom ROMAND"
      { x: 0.10, y: 0.24, w: 0.80, h: 0.06 },
      // Email
      { x: 0.10, y: 0.30, w: 0.80, h: 0.05 },
    ],
  },

  // ── Rapport mobile — client, adresse, noms techniciens
  {
    src: 'screen-rapport.png',
    dst: 'screen-rapport.png',
    regions: [
      // "Alexandra PINOT" (sous-titre)
      { x: 0.08, y: 0.02, w: 0.60, h: 0.04 },
      // Chantier "Alexandra Pl..."
      { x: 0.08, y: 0.055, w: 0.70, h: 0.05 },
      // Adresse "1294 chemin de la Bou..."
      { x: 0.08, y: 0.10, w: 0.70, h: 0.04 },
      // "Clément HEDON"
      { x: 0.04, y: 0.32, w: 0.55, h: 0.04 },
      // "Benoît MONTESINOS"
      { x: 0.04, y: 0.60, w: 0.60, h: 0.04 },
    ],
  },

  // ── Étapes mobile — client + adresse en header
  {
    src: 'screen-etapes.png',
    dst: 'screen-etapes.png',
    regions: [
      // "Alexandra PINOT"
      { x: 0.08, y: 0.02, w: 0.60, h: 0.04 },
      // Chantier "Alexandra Pl..."
      { x: 0.08, y: 0.055, w: 0.70, h: 0.05 },
      // Adresse
      { x: 0.08, y: 0.10, w: 0.70, h: 0.04 },
    ],
  },

  // ── Planning desktop — noms colonnes + filtre + "ROMAND tom" + noms chantiers
  {
    src: 'screen-planning.png',
    dst: 'screen-planning.png',
    regions: [
      // "Vue filtrée : PVPilot" (barre orange)
      { x: 0.00, y: 0.07, w: 0.60, h: 0.05 },
      // "ROMAND tom / Admin" (coin haut droite)
      { x: 0.75, y: 0.00, w: 0.25, h: 0.07 },
      // Ligne des noms colonnes (Benoît, Clément, Jules, Mathias, Test, Tom, Vincent...)
      { x: 0.10, y: 0.23, w: 0.90, h: 0.12 },
      // Contenu cellules chantier (toutes les lignes Lun→Ven)
      { x: 0.10, y: 0.35, w: 0.90, h: 0.55 },
    ],
  },
]

// ── Run
;(async () => {
  let ok = 0
  for (const s of SCREENS) {
    const srcPath = path.join(src, s.src)
    const dstPath = path.join(dst, s.dst)

    if (!fs.existsSync(srcPath)) {
      console.warn(`⚠️  Fichier manquant (ignoré) : ${srcPath}`)
      continue
    }

    try {
      await blurRegions(srcPath, dstPath, s.regions)
      ok++
    } catch (e) {
      console.error(`❌ Erreur sur ${s.src} :`, e.message)
    }
  }

  console.log(`\n🎉 ${ok}/${SCREENS.length} écrans traités → dossier public/`)
})()
